### DISCUSSION TITLE
What's BSP?

### DISCUSSION THREAD

**Discussion by TheFern2** [stm32-rs/stm32f4xx-hal#621] [Q&A] [answered]
Sorry if this is a silly question, new to rust development 

> Check if the BSP for your board exists in the stm32-rs page. If it exists, the stm32f4xx-hal crate should be already included, so you can use the bsp as BSP for your project.

What's the meaning of this paragraph?

I have a sample setup for my nucleo https://github.com/TheFern2/test_nucleo when using cargo embed I get a warning about using custom setup for the board stm324266retx. Code works though just wondering if I'm doing something wrong.

```
 ~/D/test_nucleo-master  cargo embed --target thumbv7em-none-eabihf                              123ms
    Finished dev [unoptimized + debuginfo] target(s) in 0.05s
      Config default
      Target /home/fernandob/Downloads/test_nucleo-master/target/thumbv7em-none-eabihf/debug/blinky-rs
        WARN probe_rs::config::target > Using custom sequence for ARMv7 STM32F446RETx
     Erasing sectors ✔ [00:00:01] [#######################] 48.00 KiB/48.00 KiB @ 43.86 KiB/s (eta 0s )
 Programming pages   ✔ [00:00:00] [#######################] 38.00 KiB/38.00 KiB @ 38.64 KiB/s (eta 0s )    Finished flashing in 2.098s
        Done processing config default
```

**Comment by burrbull** [comment, marked as answer]
`BSP` is board support package. Usually small crates with additional wrappers for board. For example:
https://github.com/stm32-rs/stm32f407g-disc/

You can just ignore it.

**Comment by burrbull** [comment]
> `WARN probe_rs::config::target > Using custom sequence for ARMv7 STM32F446RETx`

This is unrelated to that phrase.
It just says that `probe-rs` (cargo-embed) can change some registers state during flashing. For example enable DBG clocking in sleep mode.
